### PR TITLE
CLOUDP-45904: Create single logger

### PR DESCRIPTION
**Pull request by mrgunior**
This PR includes changes that relates to: Using a single logger across the broker, code documentation, and the logger.go module.

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+// Session sets the session of the logger and returns a new logger with a nested session. We are currently
+// returning the same logger back.
+func (lagerZapLogger *LagerZapLogger) Session(task string, data ...lager.Data) lager.Logger {
+	return lagerZapLogger
+}
+
+// WithData creates a new child with the parent fields and returns a logger with newly added data. We are currently
+// returing the same logger back.
```
Comment (BenElgar) [inline logger.go]:
```suggestion
// returning the same logger back.
```

File: logger.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+// Fatal logs a message at FatalLevel. The message includes any fields passed
+// at the log site, as well as any fields accumulated on the logger.
+func (lagerZapLogger *LagerZapLogger) Fatal(action string, err error, data ...lager.Data) {
+	lagerZapLogger.logger.Fatalw(action, createFields(data)...)
+}
+
+// createFields converts the structured log data that the lager library uses to
+// the format that zap expects. lager uses a list of maps while zap expects a flat list
+// of alternating keys and values.
+func createFields(data []lager.Data) []interface{} {
```
Comment (BenElgar) [inline logger.go]: Could save a line by using a named return.

File: main.go [hunk 3]
```diff
@@ -5,13 +5,13 @@ import (
 	"net/http"
 	"strconv"
 
+	"go.uber.org/zap"
```
Comment (BenElgar) [inline main.go]: Don't reorder the imports.

File: logger.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+func NewLagerZapLogger(zap *zap.SugaredLogger) *LagerZapLogger {
+	return &LagerZapLogger{
+		logger: zap,
+	}
+}
+
+// RegisterSink represents a write destination for a Logger. It provides a thread-safe interface for writing logs.
+func (lagerZapLogger *LagerZapLogger) RegisterSink(sink lager.Sink) {}
+
+// SessionName returns the name of the session. This is normally added when initializing a new logger but zap doesn't require nor need it.
```
Comment (BenElgar) [inline logger.go]: This whole comment doesn't make much sense to me.

File: main.go (hunk 3 above)
Comment (mrgunior) [reply main.go]: I think this is probably visual studio code doing it, unless I forgot I did this myself. :)

File: logger.go (hunk 4 above)
Comment (mrgunior) [reply logger.go]: This comment was taken from the official docs. I haven't really gone in depth to see what the sink does, seeing that zap doesn't require it. I left it so.

File: logger.go (hunk 1 above)
Comment (mrgunior) [reply logger.go]: I'll add this change. I presume seeing that these are mostly comments I can push directly to master?

File: logger.go (hunk 2 above)
Comment (mrgunior) [reply logger.go]: I'll add this change. I presume seeing that these are mostly comments I can push directly to master?

**Pull request merged**